examples/basic/index.android.js

```diff
@@ -30,8 +30,45 @@ class VideoPlayer extends Component {
 
   onLoad = (data) => {
     this.setState({ duration: data.duration });
+    this.onAudioTracks(data)
+    this.onTextTracks(data)
   };
 
+  onAudioTracks = (data: any) => {
+    const selectedTrack = data.audioTracks?.find((x: any) => {
+      return x.selected
+    })
+    this.setState({
+      audioTracks: data.audioTracks,
+    })
+    if (selectedTrack?.language) {
+      this.setState({
+        selectedAudioTrack: {
+          type: 'language',
+          value: selectedTrack?.language,
+        },
+      })
+    }
+  }
+  onTextTracks = (data: any) => {
+    const selectedTrack = data.textTracks?.find((x: any) => {
+      return x.selected
+    })
+
+    this.setState({
+      textTracks: data.textTracks,
```

Comment:
how come you call setState twice if the selected track has language? would have expected an if..else

Reply:
probably the original author created this funky example 😆 when we consume this, we wouldnt be doing a setState at all. we will utilise the logic from `onLoad` and apply the same to `onTextTracks`

Reply:
oh sorry totally didnt see this was just example code